modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/delegate/delegate.es.js

```diff
@@ -35,11 +35,16 @@ function isDisabled(node) {
  */
 function delegate(element, eventName, selector, callback) {
 	const eventHandler = (event) => {
-		if (eventName === 'click' && isDisabled(event.target)) {
+		const {target, defaultPrevented} = event;
+
+		if (eventName === 'click' && isDisabled(target)) {
 			return;
 		}
 
-		if (!event.defaultPrevented && event.target.matches(selector)) {
+		if (
+			!defaultPrevented &&
+			(target.matches(selector) || target.closest(selector))
```

Comment:
@jbalsas @matuzalemsteles @wincent 

Do any of you know where this logic is handled within `metal-dom`? I was looking through the implementation of delegate there, and can't seem to figure out how they achieve this behavior without calling `closest`. Code I was looking at is [here](https://github.com/metal/metal.js/blob/5b5032dd840bc3364894776c8bb2fb82517e2553/packages/metal-dom/src/domNamed.js#L928)

I also was using [this codesandbox](https://codesandbox.io/s/trusting-gates-17le0?file=/index.html) to test

Reply:
calling `.closest` may be the best way to achieve this, but I didn't want to just add that without seeing how metal-dom was doing it

Reply:
Is it coming via?

- [`isAbleToInteractWith()`](https://github.com/metal/metal.js/blob/5b5032dd840bc3364894776c8bb2fb82517e2553/packages/metal-dom/src/domNamed.js#L286)
- which [calls](https://github.com/metal/metal.js/blob/5b5032dd840bc3364894776c8bb2fb82517e2553/packages/metal-dom/src/domNamed.js#L302) `parent()`
- which [itself calls `closest()`](https://github.com/metal/metal.js/blob/5b5032dd840bc3364894776c8bb2fb82517e2553/packages/metal-dom/src/domNamed.js#L540-L542)

(`isAbletoInteractWith()` is called from [`triggerDelegatedListeners_()`](https://github.com/metal/metal.js/blob/5b5032dd840bc3364894776c8bb2fb82517e2553/packages/metal-dom/src/domNamed.js#L746)).

Based on a quick scan, so don't take my word for it...

Reply:
ah, I think I may just be blind. Your comment made me re-look and noticed this line immediately https://github.com/metal/metal.js/blob/5b5032dd840bc3364894776c8bb2fb82517e2553/packages/metal-dom/src/domNamed.js#L756

In that case, I think calling `.closest` should be fine here.

@kresimir-coko can you add some unit tests to account for this use case?

Reply:
Yes, I can add some unit tests, of course.

@jbalsas should I split this into 2 tasks on JIRA, one for the improvements to the implementations, and the other for usages?

Reply:
> @jbalsas should I split this into 2 tasks on JIRA, one for the improvements to the implementations, and the other for usages?

There's already a task in JIRA for that. Just use that LPS number